addons/PlayerMapMarkers/scripts/Game/FPM/Map/Markers/Config/FPM_MapMarkerEntryPlayer.c

```diff
@@ -74,6 +74,9 @@ class FPM_MapMarkerEntryPlayer : SCR_MapMarkerEntryDynamic
 			return;
 		
 		playerController.m_FPM_OnControlledEntityChangedServer.Insert(OnControlledEntityChangedServer);
+
+		// Force initial visibility check for players reconnecting
+		SCR_MapMarkerManagerComponent.GetInstance().SetStreamRulesForPlayer(playerId);
```

Comment:
This part should be removed as, it's too early. The base game code already calls `SetStreamRulesForPlayer` when the player gets a faction assigned.